From 86f69fe9c069dd8608d238581eea259caa1dfc99 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Mon, 13 Oct 2008 14:32:52 -0700
Subject: [PATCH] Staging: workaround build system bug

This is needed as CONFIG_STAGING is set to y, yet there is no code in
drivers/staging/ to build, so the build-in.o doesn't get created
properly.  Create a "dummy" module in drivers/staging called staging.c
to work around this bug.

Cc: Sam Ravnborg <sam@ravnborg.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/Makefile  |    3 +++
 drivers/staging/staging.c |   19 +++++++++++++++++++
 2 files changed, 22 insertions(+), 0 deletions(-)
 create mode 100644 drivers/staging/staging.c

--- a/drivers/staging/Makefile
+++ b/drivers/staging/Makefile
@@ -1,5 +1,8 @@
 # Makefile for staging directory
 
+# fix for build system bug...
+obj-$(CONFIG_STAGING)		+= staging.o
+
 obj-$(CONFIG_ET131X)		+= et131x/
 obj-$(CONFIG_VIDEO_GO7007)	+= go7007/
 obj-$(CONFIG_USB_IP_COMMON)	+= usbip/
--- /dev/null
+++ b/drivers/staging/staging.c
@@ -0,0 +1,19 @@
+#include <linux/kernel.h>
+#include <linux/init.h>
+#include <linux/module.h>
+
+static int __init staging_init(void)
+{
+	return 0;
+}
+
+static void __exit staging_exit(void)
+{
+}
+
+module_init(staging_init);
+module_exit(staging_exit);
+
+MODULE_AUTHOR("Greg Kroah-Hartman");
+MODULE_DESCRIPTION("Staging Core");
+MODULE_LICENSE("GPL");
